gensim/models/phrases.py

```diff
@@ -227,12 +243,17 @@ def export_phrases(self, sentences, out_delimiter=b' ', as_tuples=False):
             then you can debug the threshold with generated tsv
         """
         for sentence in sentences:
-            s = [utils.any2utf8(w) for w in sentence]
             last_bigram = False
             vocab = self.vocab
             threshold = self.threshold
             delimiter = self.delimiter  # delimiter used for lookup
             min_count = self.min_count
+
+            if self.recode_to_utf8:
+                s = [utils.any2utf8(w) for w in sentence]
+            else:
+                s = [utils.any2utf8(w) for w in sentence] if self.is_input_bytes else list(sentence)
```

Comment:
Yes, I see what you mean.

Let's see what the devs think, but it's a subtle point, I can see arguments for both ways (convert, fail).

Other than that this PR looks good, nice job @prakhar2b 👍